File: Algorithms/defanging-an-ip-address.cpp
// Link: https://leetcode.com/problems/defanging-an-ip-address/

class Solution {
public:
    string defangIPaddr(string address) {
        string result = "";
        int len = address.size();
        for(int i = 0; i < len; i++)
            if(address[i] == '.') result = result + "[.]";
            else result = result + address[i];
        return result;
    }
};

File: Algorithms/defanging-an-ip-address.swift
// Link : https://leetcode.com/problems/defanging-an-ip-address

class Solution {
    func defangIPaddr(_ address: String) -> String {
        var ret = String();
        for (index, char) in address.enumerated() {
            if(char == ".") {
                ret = ret + "[.]";
            }
            else {
                ret = ret + String(char);
            }
        }
        return ret;
    }
}
